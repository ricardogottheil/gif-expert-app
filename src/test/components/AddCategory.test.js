import React from 'react';
import '@testing-library/jest-dom';

const { shallow } = require('enzyme');
const { AddCategory } = require('../../components/AddCategory');

describe('Puebas con AddCategory', () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('Debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de cambiar la caja de texto', () => {
    const input = wrapper.find('input');
    const value = 'Hola Mundo';
    input.simulate('change', {
      target: {
        value,
      },
    });
    expect(wrapper.find('p').text().trim()).toBe(value);
  });

  test('No debe postear la informaciÃ³n con Submit', () => {
    wrapper.find('form').simulate('submit', {
      preventDefault() {},
    });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test('Debe llamar el setCategories y limpiar la caja de texto', () => {});
});
