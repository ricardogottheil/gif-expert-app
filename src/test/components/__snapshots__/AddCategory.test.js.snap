// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas con AddCategory Debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
